src/views/messages-view.jsx

```diff
@@ -793,6 +836,125 @@ class MessagesView extends React.Component {
     uploader.cancel();
   }
 
+  handlePickReply(m) {
+    this.setState({reply: null});
+    if (m) {
+      let cont = m.content
+      if (cont) {
+        if (typeof cont == 'string') {
+          cont = Drafty.init(cont);
+        }
+        cont = Drafty.replyPreview(cont, 30, quotePreviewFmt);
+
+        // Get the author.
+        let header;
+        const seq = m.seq;
+        const topic = this.props.tinode.getTopic(this.state.topic);
+        const m0 = topic.findMessage(seq);
+        let thisFrom;
+        if (m0) {
+          thisFrom = m0.from || 'chan';
+          if (thisFrom != this.props.myUserId) {
+            const topic = this.props.tinode.getTopic(this.state.topic);
+            const user = topic.userDesc(thisFrom);
+            if (user && user.public) {
+              header = user.public.fn;
+            }
+          } else {
+            header = this.props.myUserName;
+          }
+        }
+
+        header = Drafty.init(header || '');
+        // Make small image previews.
+        const ents = [];
+        if (cont.ent) {
+          Drafty.entities(cont, (data, idx, tp) => {
+            if (tp == 'IM') {
+              ents.push({
+                tp: tp,
+                data: data
+              });
+            }
+
+          }, this);
+        }
+        this.convertIntoThumbnails(ents, 0, (success) => {
+          if (success) {
+            const msg = Drafty.createQuote(header, cont, letterTileColorId(thisFrom));
+            this.setState({reply: {content: msg, seq: m.seq}});
+          }
+        });
+
+        return;
+      }
+    }
+  }
+
+  // Turn images in the provided Drafty.ent entries into thumbnails.
+  convertIntoThumbnails(ents, idx, done) {
+    if (idx >= ents.length) {
+      done(true);
+      return;
+    }
+
+    const scale = (origBlob) => {
+      imageScaled(origBlob, IMAGE_THUMBNAIL_DIM, IMAGE_THUMBNAIL_DIM, -1, false,
+        // Success
+        (mime, blob, width, height, fname) => {
+          const ex = ents[idx];
```

Comment:
This looks a bit fishy. `ex` is a reference to the original object stored in ents[idx]. Then below you are modifying the original through the reference. Please double check that it's the intended behavior.